### PR TITLE
[AutoDiff] NFC: prettify commutative diagrams.

### PR DESCRIPTION
**Pull request by dan-zheng**
Make commutative diagrams pretty using Unicode box characters.
Use lowercase letters for arrow names.

---

Rationale: if we're going to draw a diagram, we might as well make it as pretty as possible. Maintenance shouldn't be an issue.

Unicode resource: https://antofthy.gitlab.io/info/data/utf8-demo.txt

Inspired by: https://twitter.com/lexi_lambda/status/1255988716390297601

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please smoke test

**Review by rxwei — APPROVED** [review]

File: lib/SIL/IR/SILFunctionType.cpp [hunk 1]
```diff
@@ -3088,30 +3088,31 @@ TypeConverter::getConstantInfo(TypeExpansionContext expansion,
...
   //   lowering it via SILGen. This is the default lowering behavior, but may
   //   break SIL typing invariants because expected lowered derivative types are
   //   computed from lowered original function types.
-  // - (AD) is the result of lowering the original function type, then computing
+  // - (ad) is the result of lowering the original function type, then computing
   //   its derivative type. This is the expected lowered derivative type,
   //   preserving SIL typing invariants.
   //
-  // Always use (AD) to compute lowered derivative function types.
+  // Always use (ad) to compute lowered derivative function types.
```
Comment (dan-zheng) [inline lib/SIL/IR/SILFunctionType.cpp:3115]: Nice autodiff pun: `Always use (ad)`

**Comment by dan-zheng** [comment]
Linux CI seems to have flaked.
@swift-ci Please smoke test Linux

**Comment by dan-zheng** [comment]
@swift-ci Please smoke test Linux

**Comment by dan-zheng** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**